go/decompress/decompress.go

```diff
@@ -0,0 +1,205 @@
+package decompress
+
+import (
+	"archive/tar"
+	"bufio"
+	"fmt"
+	"io"
+	"io/fs"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/biogo/hts/bgzf"
+	gzip "github.com/klauspost/pgzip"
+)
+
+const EXPECTED_ANNOTATION_FILE_SUFFIX = "annotation.tsv.gz"
+
+type BystroReader interface {
+	ReadLines() ([]byte, error)
+	ReadLine() ([]byte, error)
+}
+
+type BzfBystroReader struct {
+	Reader *bgzf.Reader
+}
+
+type BufioBystroReader struct {
+	Reader *bufio.Reader
+}
+
+// Read a line up to the next newline character, and return the line excluding the newline character
+// Implementation follows the example in the bgzf package documentation: https://github.com/biogo/hts/blob/bb1e21d1bfc7f2b1e124ca0a1ed98493d191db78/bgzf/line_example_test.go#L70
+func _readLineBgzip(r *bgzf.Reader) ([]byte, error) {
```

Comment:
If you're aiming for idomatic go I'd 1) drop the underscore from the function name and 2) add a comment that begins // readLineBgzip takes a reader and returns a byte slice that ends in a new line or an error.